homeassistant/components/mqtt/device_tracker.py

```diff
@@ -151,16 +156,44 @@ def _process_update_extra_state_attributes(
         self, extra_state_attributes: dict[str, Any]
     ) -> None:
         """Extract the location from the extra state attributes."""
-        self._attr_latitude = extra_state_attributes.get(ATTR_LATITUDE)
-        self._attr_longitude = extra_state_attributes.get(ATTR_LONGITUDE)
         if (
             ATTR_LATITUDE in extra_state_attributes
             or ATTR_LONGITUDE in extra_state_attributes
         ):
-            # Reset manual set location
+            latitude: float | None
+            longitude: float | None
+            gps_accuracy: int
+            # Reset manually set location to allow automatic zone detection
             self._attr_location_name = None
+            if isinstance(
+                latitude := extra_state_attributes.get(ATTR_LATITUDE), (int, float)
+            ) and isinstance(
+                longitude := extra_state_attributes.get(ATTR_LONGITUDE), (int, float)
+            ):
+                self._attr_latitude = latitude
+                self._attr_longitude = longitude
+            else:
+                # Invalid or incomplete coordinates, reset location
+                self._attr_latitude = None
+                self._attr_longitude = None
+                _LOGGER.warning(
+                    "Extra state attributes received at % and template %s "
+                    "contain invalid or incomplete location info. Got %s",
+                    self._config.get(CONF_JSON_ATTRS_TEMPLATE),
+                    self._config.get(CONF_JSON_ATTRS_TOPIC),
+                    extra_state_attributes,
+                )
+
+            self._attr_location_accuracy = (
+                gps_accuracy
+                if ATTR_GPS_ACCURACY in extra_state_attributes
+                and isinstance(
+                    gps_accuracy := extra_state_attributes[ATTR_GPS_ACCURACY],
+                    (int, float),
+                )
+                else 0
```

Comment:
Why does this fall back to `0` instead of logging a warning and setting it to `None` like we do for the coordinates?

Reply:
`_attr_gps_accuracy` must be an `int` according to the typing and is not allowed to be `None`

https://github.com/home-assistant/core/blob/c7ca88e666643fde92d61ffabdcedafadbb351ff/homeassistant/components/device_tracker/config_entry.py#L214-L224

Logging was added

Reply:
Oh, I see. I think we should allow it to be None, but in a separate architecture issue.